File: examples/python/subagent.py
"""Example: Enable subagent support and let the model delegate tasks."""

import asyncio
import os

from agentlib import Agent, AgentConfig


async def main():
    config = AgentConfig(
        api_key=os.environ.get("ANTHROPIC_AUTH_TOKEN", ""),
        model=os.environ.get("ANTHROPIC_MODEL", "claude-sonnet-4-6"),
        enable_subagent=True,
        subagent_max_turns=20,
    )

    agent = Agent(config)

    async for event in agent.run(
        "Explore this codebase. Use subagents to parallelize exploration of "
        "different directories, then combine the findings."
    ):
        if event.type.value == "message_delta":
            print(event.data["text"], end="", flush=True)
        elif event.type.value == "complete":
            print(f"\n\n[Complete] {event.data['final_content']}")
        elif event.type.value == "error":
            print(f"\n[Error] {event.data['message']}")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/nodejs/subagent.ts
/** Example: Enable subagent support and let the model delegate tasks. */

import { Agent, resolveConfig } from '../src/index.js';

async function main() {
  const config = resolveConfig({
    apiKey: process.env.ANTHROPIC_AUTH_TOKEN || '',
    enableSubagent: true,
    subagentMaxTurns: 20,
  });

  const agent = new Agent(config);

  for await (const event of agent.run(
    'Explore this codebase using subagents to parallelize exploration'
  )) {
    if (event.type === 'message_delta') {
      process.stdout.write(event.data.text as string);
    } else if (event.type === 'complete') {
      console.log('\n\n[Complete]', event.data.final_content);
    } else if (event.type === 'error') {
      console.error('\n[Error]', event.data.message);
    }
  }
}

main().catch(console.error);
